packages/material-ui-lab/src/TimelineContent/TimelineContent.js

```diff
@@ -1,39 +1,67 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
+import { deepmerge } from '@material-ui/utils';
 import { capitalize } from '@material-ui/core/utils';
-import { withStyles } from '@material-ui/core/styles';
+import { experimentalStyled } from '@material-ui/core/styles';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import useThemeProps from '@material-ui/core/styles/useThemeProps';
 import Typography from '@material-ui/core/Typography';
 import TimelineContext from '../Timeline/TimelineContext';
 import TimelineItemContext from '../TimelineItem/TimelineItemContext';
+import { getTimelineContentUtilityClass } from './timelineContentClasses';
 
-export const styles = () => ({
-  /* Styles applied to the root element. */
-  root: {
-    flex: 1,
-    padding: '6px 16px',
-  },
-  /* Styles applied to the root element if `align="right"`. */
-  alignRight: {
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+  return deepmerge(
+    {
+      ...styleProps[`align${capitalize(align)}`],
+    },
+    styles.root || {},
+  );
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { align, classes } = styleProps;
+
+  const slots = {
+    root: ['root', `align${capitalize(align)}`],
+  };
+
+  return composeClasses(slots, getTimelineContentUtilityClass, classes);
+};
+
+const TimelineContentRoot = experimentalStyled(
+  Typography,
+  {},
+  { name: 'MuiTimelineContent', slot: 'Root', overridesResolver },
+)(({ styleProps }) => ({
+  flex: 1,
+  padding: '6px 16px',
+  ...(styleProps.align === 'right' && {
     textAlign: 'right',
-  },
-});
+  }),
+}));
 
-const TimelineContent = React.forwardRef(function TimelineContent(props, ref) {
-  const { classes, className, ...other } = props;
+const TimelineContent = React.forwardRef(function TimelineContent(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiTimelineContent' });
+  const { className, ...other } = props;
 
   const { align = 'left' } = React.useContext(TimelineContext);
   const { classes: contextClasses = {} } = React.useContext(TimelineItemContext);
 
+  const styleProps = {
+    align,
+    ...other,
+  };
```

Comment:
```suggestion
  const styleProps = {
    ...props,
    align,
  };
```

For consistency with the other components